src/nupic/algorithms/Connections.hpp

```diff
@@ -425,10 +432,12 @@ class Connections : public Serializable
    * @param increment  Change in permanence for synapses with active presynapses.
    * @param decrement  Change in permanence for synapses with inactive presynapses.
    */
-  void adaptSegment(const Segment segment, 
+  void adaptSegment(const Segment segment,
                     const sdr::SDR &inputs,
                     const Permanence increment,
-                    const Permanence decrement);
+                    const Permanence decrement,
+                    const std::vector<Permanence> &previousUpdates,
```

Comment:
known- TODO doc

Reply:
I added documentation for the parameters.  See if you can make sense of it?  I think it explains several of your questions.